### PR TITLE
Feat/RTENU-192 filelist frontend

### PR DESCRIPTION
**Pull request by saneDG**
- Add components for adding/removing lists in frontend
- Fixes to endpoints that are used CRUD list component

These changes should not affect user in any way yet. Some changes are still needed before we can use these changes. You should be able to test component by uncommenting [packages/frontend/src/pages/ProtectedPage/index.tsx](https://github.com/finnishtransportagency/ratatiedot-extranet/compare/feat/RTENU-192-filelist-frontend?expand=1#diff-88e85b5fd3a792fc51ac93cdfdf0b240bc8ee28cd4c1cb4975a11d1b5999e580) line 76.

### CODE REVIEW

File: packages/frontend/src/pages/Files/Files.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Typography } from '@mui/material';
```
Comment (NinaDang97) [inline packages/frontend/src/pages/Files/Files.tsx:1]: How do you plan to use this component?

File: packages/server/lambdas/alfresco/delete-folder.ts [hunk 2]
```diff
@@ -61,13 +60,9 @@ export async function handleRequest(event: ALBEvent): Promise<ALBResult | undefi
     validateWriteUser(user, writeRole);
 
     const headers = (await getAlfrescoOptions(user.uid)).headers;
-    const requestOptions = folderDeleteRequestBuilder(headers) as RequestInit;
-
-    const alfrescoResult = await deleteFolder(requestOptions, nodeId);
-    if (!alfrescoResult) {
-      throw new RataExtraLambdaError('Error deleting folder from Alfresco', 404);
-    }
+    const requestOptions = (await folderDeleteRequestBuilder(headers)) as RequestInit;
```
Comment (NinaDang97) [inline packages/server/lambdas/alfresco/delete-folder.ts:64]: Could I ask why did you remove conditional check`!alfrescoResult` before deleting database record?

File: packages/server/lambdas/alfresco/delete-folder.ts [hunk 3]
```diff
@@ -76,9 +71,8 @@ export async function handleRequest(event: ALBEvent): Promise<ALBResult | undefi
     auditLog.info(user, `Deleted folder ${nodeId} in ${categoryData.alfrescoFolder}`);
 
     return {
-      statusCode: 200,
+      statusCode: 204,
```
Comment (NinaDang97) [inline packages/server/lambdas/alfresco/delete-folder.ts:74]: Hmm why not `statusCode: 200`?

**Review by NinaDang97 — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/Modal/ListModal.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  title?: string;
+  children?: ReactElement<any, string | JSXElementConstructor<any>>;
+  open: boolean;
+  error?: boolean;
+  success?: boolean;
+  handleClose: () => void;
+  onSnackbarClose?: () => void;
+}
+
+export const ListModal: FunctionComponent<ModalProps> = ({
```
Comment (NinaDang97) [inline packages/frontend/src/components/Modal/ListModal.tsx:22]: Is ListModal component duplicated version of FileModal? I can't see their differences yet

File: packages/frontend/src/pages/Files/Files.tsx (hunk 1 above)
Comment (saneDG) [reply packages/frontend/src/pages/Files/Files.tsx:1]: We would use this page for listing the files of single alfresco folder. Since this does not do anything right now I guess we dont need it in this PR.

File: packages/server/lambdas/alfresco/delete-folder.ts (hunk 3 above)
Comment (saneDG) [reply packages/server/lambdas/alfresco/delete-folder.ts:74]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204

File: packages/server/lambdas/alfresco/delete-folder.ts (hunk 2 above)
Comment (saneDG) [reply packages/server/lambdas/alfresco/delete-folder.ts:64]: Good catch. I remember the conditional causing some issues when deleting folder and I must have forgotten to apply it back.

File: packages/frontend/src/components/Modal/ListModal.tsx (hunk 4 above)
Comment (saneDG) [reply packages/frontend/src/components/Modal/ListModal.tsx:22]: There are differences in some messages. We could pass those as props but I'm afraid we could create too generic component with too complex logic. 🤷🏻 

Maybe it would make more sense to keep this component as generic modal component and merge those two together.

Comment (NinaDang97) [reply packages/frontend/src/components/Modal/ListModal.tsx:22]: I also have same thinking that should merge these 2 modal components and re-name this component something more generic, e.g. ItemUploadModal, ContentUploadModal or EntityUploadModal which is more generic, could cover both file upload and list title name.

File: packages/frontend/src/components/FolderList/Folders.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        },
+      });
+      setOpen(false);
+      setSuccess(true);
+      getComponents();
+      return response;
+    } catch (error) {
+      setError(true);
+    }
+    resetState();
```
Comment (NinaDang97) [inline packages/frontend/src/components/FolderList/Folders.tsx:64]: Do we need an empty function here?

**Review by NinaDang97 — CHANGES_REQUESTED** [review]
Let's wait for Jiri or Niko's review too

File: packages/frontend/src/pages/ProtectedPage/index.tsx [hunk 6]
```diff
@@ -74,6 +75,7 @@ export const ProtectedPage = ({ children }: Props) => {
           <PageTitle routerName={categoryRouteName} />
           {categoryRouteName ? isFavorite ? <RemoveFavoriteButton /> : <AddFavoriteButton /> : <></>}
           {isEditorOpened && <SlateInputField />}
+          {/* <Folders isEditing={openEdit} /> */}
```
Comment (mehtis) [inline packages/frontend/src/pages/ProtectedPage/index.tsx:78]: Is this still needed for something later on?

File: packages/frontend/src/components/FolderList/FolderList.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+interface FolderListProps {
+  parentNode: Node;
+  isEditing: boolean;
+  title: string;
+  onEdit: (node: Node) => void;
+}
+
+export const FolderList = ({ parentNode, isEditing, title, onEdit }: FolderListProps) => {
+  const [folders, setFolders] = useState<any[]>([]);
```
Comment (mehtis) [inline packages/frontend/src/components/FolderList/FolderList.tsx:27]: Typing for this? Also the folders.map() on line 63.

File: packages/frontend/src/components/FolderList/FolderList.tsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  const getFolders = async () => {
+    try {
+      const response: any = await axios.get(`api/alfresco/nodes/${parentNode.alfrescoNodeId}?type=folder`);
```
Comment (mehtis) [inline packages/frontend/src/components/FolderList/FolderList.tsx:30]: Typing

File: packages/frontend/src/components/FolderList/Folders.tsx [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+  const [success, setSuccess] = useState(false);
+  const [error, setError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [selectedComponent, setSelectedComponent] = useState<Component | null>(null);
+  const [isNewList, setIsNewList] = useState(false);
+  const categoryName = getCategoryRouteName(useLocation());
+
+  const getComponents = async () => {
+    try {
+      const response: any = await axios.get(`/api/database/components/${categoryName}`);
```
Comment (mehtis) [inline packages/frontend/src/components/FolderList/Folders.tsx:34]: Typing here and line below.

File: packages/frontend/src/components/FolderList/Folders.tsx [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+
+  useEffect(() => {
+    getComponents();
+  }, []);
+
+  const resetState = () => {};
+
+  const addList = async (title: string) => {
+    try {
+      const response = await axios.post(`/api/alfresco/folder/${categoryName}`, {
```
Comment (mehtis) [inline packages/frontend/src/components/FolderList/Folders.tsx:49]: General idea, not just here: Could this beginning bit be in a consts-type of file? It repeats quite often in slight variations. Not something that needs to be fixed here, but something to think about.

File: packages/server/lambdas/alfresco/list-nodes.ts [hunk 11]
```diff
@@ -1,12 +1,12 @@
...
 import axios from 'axios';
-import { getAlfrescoUrlBase } from '../../utils/alfresco';
+import { getAlfrescoOptions, getAlfrescoUrlBase } from '../../utils/alfresco';
 
 import { getRataExtraLambdaError, RataExtraLambdaError } from '../../utils/errors';
 import { log } from '../../utils/logger';
 import { getUser, validateReadUser } from '../../utils/userService';
 
-const getNodes = async (id: string, type?: string) => {
+const getNodes = async (id: string, options: any, type?: string) => {
```
Comment (mehtis) [inline packages/server/lambdas/alfresco/list-nodes.ts:9]: Any chance to do some sort of typing here? Even `Record<string, string>`?

**Review by mehtis — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/FolderList/Folders.tsx [hunk 12]
```diff
@@ -61,10 +58,9 @@ export const Folders = ({ isEditing }: FoldersProps) => {
     } catch (error) {
       setError(true);
     }
-    resetState();
   };
 
-  const editList = async (component: Component | null, title: string) => {
+  const editList = async (component: Node | null, title: string) => {
```
Comment (NinaDang97) [inline packages/frontend/src/components/FolderList/Folders.tsx:63]: What's the difference between `Component` and `Node` type here and packages/frontend/src/components/FolderList/Folders.tsx line 13?

File: packages/frontend/src/components/Files/FileUploadDialog.tsx [hunk 13]
```diff
@@ -85,10 +85,12 @@ export const FileUploadDialog = ({ categoryName, open, onClose, onUpload }: File
   switch (dialogPhase) {
     case 1:
       return (
-        <FileModal
+        <Modal
           open={open}
           onSnackbarClose={handleSnackbarClose}
           handleClose={handleClose}
+          errorMessage={t('common:file.files_not_deleted')}
+          successMessage={t('common:file.files_deleted')}
```
Comment (noravux) [inline packages/frontend/src/components/Files/FileUploadDialog.tsx:93]: Change the translations to be about uploading the file, not deleting?

File: packages/frontend/src/components/FolderList/Folders.tsx (hunk 12 above)
Comment (saneDG) [reply packages/frontend/src/components/FolderList/Folders.tsx]: fixed

**Review by NinaDang97 — APPROVED** [review]

**Review by noravux — APPROVED** [review]

File: packages/frontend/src/components/FolderList/FolderList.tsx [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+        </Typography>
+        <IconButton
+          sx={{ color: Colors.extrablack, marginRight: '-10px', marginLeft: 'auto' }}
+          aria-label={t('common:edit.edit')}
+          onClick={() => onEdit(parentNode)}
+        >
+          <Edit />
+        </IconButton>
+      </Box>
+      {folders.map((folder: any) => (
```
Comment (mehtis) [inline packages/frontend/src/components/FolderList/FolderList.tsx:63]: Typing?

File: packages/server/lambdas/alfresco/list-nodes.ts [hunk 15]
```diff
@@ -16,9 +16,10 @@ const getNodes = async (id: string, type?: string) => {
     if (type && type === 'file') {
       queryParameter = `?where=(isFile=true)`;
     }
-    return await axios.get(`${alfrescoCoreAPIUrl}/nodes/${id}/children${queryParameter}`);
+    const response = await axios.get(`${alfrescoCoreAPIUrl}/nodes/${id}/children${queryParameter}`, options);
+    return response;
   } catch (error) {
-    return error;
+    log.error(error);
```
Comment (mehtis) [inline packages/server/lambdas/alfresco/list-nodes.ts:22]: No complaints from linter or anything that we're not returning anything here? One possibility would be to remove the try-catch here and let any errors be caught in the handleRequest try-catch. This is something that probably should be applied elsewhere as well... 🤔

File: packages/server/lambdas/alfresco/delete-folder.ts [hunk 16]
```diff
@@ -8,20 +8,18 @@ import { DatabaseClient } from '../database/client';
...
+const deleteFolder = async (options: RequestInit, nodeId: string) => {
   const alfrescoCoreAPIUrl = `${getAlfrescoUrlBase()}/alfresco/versions/1`;
   const url = `${alfrescoCoreAPIUrl}/nodes/${nodeId}`;
   try {
     const res = await fetch(url, options);
-    const result = (await res.json()) as AlfrescoResponse;
-    return result;
+    return res;
   } catch (err) {
     console.error('error:' + err);
```
Comment (mehtis) [inline packages/server/lambdas/alfresco/delete-folder.ts:24]: Happened to notice while checking this other error stuff, that it's using console.error, not log.error.

**Review by mehtis — APPROVED** [review]
